ci/build_setup.sh

```diff
@@ -83,3 +79,17 @@ cp -f "${ENVOY_SRCDIR}"/ci/WORKSPACE "${ENVOY_BUILD_DIR}"
 # This is where we copy build deliverables to.
 export ENVOY_DELIVERY_DIR="${ENVOY_BUILD_DIR}"/source/exe
 mkdir -p "${ENVOY_DELIVERY_DIR}"
+
+# This is where we build for bazel.release* and bazel.dev.
+export ENVOY_CI_DIR="${ENVOY_SRCDIR}"/ci
+
+# Hack due to https://github.com/lyft/envoy/issues/838 and the need to have
+# tools and bazel.rc available for build linkstamping.
+mkdir -p "${ENVOY_CONSUMER_SRCDIR}"/tools
+mkdir -p "${ENVOY_CI_DIR}"/tools
+ln -sf "${ENVOY_SRCDIR}"/tools/bazel.rc "${ENVOY_CONSUMER_SRCDIR}"/tools/
+ln -sf "${ENVOY_SRCDIR}"/tools/bazel.rc "${ENVOY_CI_DIR}"/tools/
+mkdir -p "${ENVOY_CONSUMER_SRCDIR}"/bazel
+mkdir -p "${ENVOY_CI_DIR}"/bazel
```

Comment:
Actually, one more thing.

Could you remove `ci/{bazel,tools}` directories? or if you don't think that's valid option, then add them to `.gitignore`? Otherwise tree isn't clean.